server/src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -665,14 +665,16 @@ private static DiscoveryNode findSourceNodeForPeerRecovery(Logger logger, Routin
     private class RecoveryListener implements PeerRecoveryTargetService.RecoveryListener {
 
         private final ShardRouting shardRouting;
+        private final long primaryTerm;
```

Comment:
add a comment that this is the shard routing and primary term with which the shard was created.

Reply:
Ok